rust/connlib/tunnel/src/peer.rs

```diff
@@ -271,37 +271,24 @@ impl ClientOnGateway {
     }
 
     pub(crate) fn handle_timeout(&mut self, now: Instant) {
-        let expired_translations = self
+        for (proxy_ip, state) in self
             .permanent_translations
             .iter()
-            .filter(|(_, state)| state.slated_for_refresh);
-        let mut for_refresh = HashSet::new();
-        for (proxy_ip, expired_state) in expired_translations {
-            if self
-                .permanent_translations
-                .values()
-                .filter_map(|state| {
-                    (state.resource_id == expired_state.resource_id
-                        && state.name == expired_state.name)
-                        .then_some((state.last_response, state.slated_for_refresh))
-                })
-                .all(|(last_seen, slated_for_refresh)| {
-                    slated_for_refresh || now.duration_since(last_seen) > Duration::from_secs(30)
-                })
-            {
```

Comment:
I am not sure why this was done in this way, it appears to me that a simple `.filter` has the same effect? Maybe I am missing something.